### PR TITLE
feat(registry): add remove latest partition job

### PR DESCRIPTION
**Pull request by bnchrch**
## What
Provides a job to delete all latest partitions.

Why?

If you remove the partitions they will be readded **triggering a reprocess of latest**
## Can this PR be safely reverted and rolled back?

- [ ] YES 💚
- [ ] NO ❌

### CODE REVIEW

**Comment by bnchrch** [comment]
* **#44795** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/44795?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @bnchrch and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: airbyte-ci/connectors/metadata_service/orchestrator/pyproject.toml
Comment (erohmensing) [inline airbyte-ci/connectors/metadata_service/orchestrator/pyproject.toml:3]:
```suggestion
version = "0.5.0"
```
Since we added a new job

File: airbyte-ci/connectors/metadata_service/orchestrator/orchestrator/jobs/registry.py
Comment (erohmensing) [inline airbyte-ci/connectors/metadata_service/orchestrator/orchestrator/jobs/registry.py:58]:
```suggestion
    This op is responsible for removing for latest metadata partitions. (Generally used to reprocess metadata files).
```
The difference between these 2 is what made it so hard for me to comprehend for so long, so keeping it correct :D

**Review by erohmensing — APPROVED** [review]
I watched ben test this new job (and subsequent materialization) with mine own eyes. Let's do it

**Review by alafanechere — COMMENTED** [review]
Three questions:
### 1. Why do yo want to reprocess latest?
My understanding of the system is that partitions are per etag, so if latest changed new metadata partition will get added. If latest did not change I don't get why we'd want to recreate the partition and retrrigger all downstream operations.

### 2. What is scheduling the job?
I'm not seeing any code change related to scheduling this `remove_latest_metadata_partitions`. There's probably an implicit logic I'm unaware of.

### 3. Would this be useful for my [release candidate PR](https://github.com/airbytehq/airbyte/pull/44588)
I want to rematerialize an asset on blob deletion. 
The asset is a list of available release candidates.

I took an approach which I'm not super satisfied of [here](https://github.com/airbytehq/airbyte/pull/44588#discussion_r1731443057).

Your change makes me think I could take the following approach:
* Consider `release_candidate` metadata as partitions
* Make the job which is remove stale metadata partition trigger asset rematerialization on partition deletion with `AssetMaterialization(asset_key=AssetKey("my_asset"), description="Rematerializing my asset")`

Does it sound better?

**Comment by bnchrch** [comment]
@alafanechere Great questions.

@erohmensing and I added this so in the event we updated the registry entry generation logic, we could manually retrigger registry entry processing.

How we do that is simply by deleting the partition keys for all latest metadata files

as it will be picked up on the next "add partition run" and processed again.

**Comment by erohmensing** [comment]
To add - for example if we add a new (generated) field in the metadata, we want to be able to populate that info for the connectors without having to re-release them

**Comment by alafanechere** [comment]
Sounds definitely legit. @erohmensing @bnchrch will you manually run the job to delete the partitions?

**Comment by erohmensing** [comment]
@alafanechere Yep, the plan is to do that in the evening in SF so as not to create a backlog that slows connectors being published during the day!

**Comment by bnchrch** [comment]
@alafanechere  As the new DAG master Ill merge when we have your approval :)

**Comment by alafanechere** [comment]
> @alafanechere  As the new DAG master Ill merge when we have your approval :)

I'm a GitHub mobile so I don't think I can 👍 from there. But I do approve! Feel free to merge

**Pull request merged**
